subprojects/snapshots/src/main/java/org/gradle/internal/vfs/VirtualFileSystem.java

```diff
@@ -36,7 +38,13 @@ public interface VirtualFileSystem {
     /**
      * Adds the information of the snapshot at the absolute path to the VFS.
      */
-    void store(String absolutePath, FileSystemLocationSnapshot snapshot);
+    FileSystemLocationSnapshot store(String absolutePath, Supplier<FileSystemLocationSnapshot> snapshotSupplier);
```

Comment:
If `store()` fails to actually store the snapshot, the caller now has no idea about that, right? Is this a problem?

Reply:
I don't think this is a problem. The VFS is a cache after all, so if something doesn't make it into the cache the caller shouldn't care, right?